### ISSUE TITLE
Handling unicode JSON results?

### ISSUE THREAD

**Issue by tlorens** [guzzle/guzzle#2323]
**Guzzle version(s) affected**: 6.3.x

**Description**  
API returns unicode JSON.  Resulting in 
`...u'description': u'THING', u'created_by': None, u'id': 10, u'report_type_id': 15, u'iscustom': 0, u'category_id': 3, u'level': u'12', u'ports': None}...
`
I'm not sure what Guzzle is doing with the results prior to returning them for me to json_decode()

**Possible Solution**  
I can only assume there's a configuration option for this.   
1. curl from cli, I don't get the 'u' 
2. Postman I also don't get the 'u'.   
3. hand-crafted curl via PHP which also doesn't return the 'u' and I'm able to json_decode() the string. 

I feel that it has something to do with it calling __toString() when casting the result body/stream to a string.

**Comment by alexeyshockov** [comment]
I'm not sure if it's a Guzzle related bug, looks like the the server is written in Python and answers with strings in [`u'...'` Unicode notation](https://docs.python.org/2/howto/unicode.html).

**Comment by tlorens** [comment]
That was my same though as well.  Everything I found dealing with 'u' in JSON is Python based.  I don't get the entire notation though, just the 'u' as described in my original post.  I've tinkered with adding and changing headers without any luck.  I'm almost certain it has something to do with large responses and that's why I believe there's a problem with the arbitrary casting to (string) that is causing the 'u' to be included in the response.  I'm able to pass date ranges and other identifiers to limit my results and when I do this, the response doesn't contain the 'u's and parses just fine.  However, larger responses are still 'broken'.
